plot_earthquake.py

```diff
@@ -0,0 +1,100 @@
+from datetime import date
+import json
+import matplotlib.pyplot as plt
+
+
+
+def get_data():
+    """Retrieve the data we will be working with."""
+    with open('jsonfile.json','r') as f:
+        loaded_json_str=f.read()
+    load=json.loads(loaded_json_str)
+
+    return load
+
+
+def get_year(earthquake):
+    """Extract the year in which an earthquake happened."""
+    timestamp = earthquake['properties']['time']
+    # The time is given in a strange-looking but commonly-used format.
+    # To understand it, we can look at the documentation of the source data:
+    # https://earthquake.usgs.gov/data/comcat/index.php#time
+    # Fortunately, Python provides a way of interpreting this timestamp:
+    # (Question for discussion: Why do we divide by 1000?)
+    year = date.fromtimestamp(timestamp/1000).year
+    return year
+
+
+def get_magnitude(earthquake):
+    """Retrive the magnitude of an earthquake item."""
+    return earthquake["properties"]["mag"]
+    ...
+
+
+# This is function you may want to create to break down the computations,
+# although it is not necessary. You may also change it to something different.
+def get_magnitudes_per_year(earthquakes,year):
+    """Retrieve the magnitudes of all the earthquakes in a given year.
+    
+    Returns a dictionary with years as keys, and lists of magnitudes as values.
+    """
+    mag=[]
+    for i in range(len(earthquakes)):
+        quake=earthquakes[i]
+        mag.append(get_magnitude(quake))
+    yearmag={year:mag}
+    return yearmag
+
+    ...
+
+
+def plot_average_magnitude_per_year(yearquakes,year):
+    avg=[]
+    for i in range(len(year)):
+        yearmag=get_magnitudes_per_year(yearquakes[i],year[i])
+        mag=yearmag[year[i]]
+        avg_mag=sum(mag)/len(mag)
+        avg.append(avg_mag)
+    plt.plot(year,avg)
```

Comment:
Plot looks pretty, but 2012 and 2016 have no earthquakes, so the average magnitude for these years should be zero. When you do the line plot, it just 'connects the dots', giving a nonzero average magnitude for these years.

Reply:
Oh yes, many thanks for your corrections. I have fixed these problems. Your assistance has been invaluable in completing this program!